### PR TITLE
Switched CanSparksMax to Coast when Disabled

### PR DESCRIPTION
**Pull request by avakimc**
- made all CanSparksMax instance variables public
- made instances of subsystems in RobotContainer
- set each CanSparksMax instances in Robot to coast when disabled, after determining that we are in a real match

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -49,11 +50,29 @@ public void robotPeriodic() {
     CommandScheduler.getInstance().run();
   }
 
-  /** This function is called once each time the robot enters Disabled mode. */
+  /** This function is called once each time the robot enters disabled mode. */
   @Override
   public void disabledInit() {
```
Comment (balghane) [inline src/main/java/frc/robot/Robot.java:57]: Need to set them to brake in auton and teleop init

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -49,11 +50,29 @@ public void robotPeriodic() {
...
         IntakeLimelightConstants.INTAKE_LIMELIGHT_NAME); // It takes 2.5-3s on first run
+
+    if (m_robotContainer.matchState.realMatch) {
+      m_robotContainer.intake.pivotMotor.setIdleMode(IdleMode.kCoast);
+      m_robotContainer.elevator.leftMotor.setIdleMode(IdleMode.kCoast);
+      m_robotContainer.elevator.rightMotor.setIdleMode(IdleMode.kCoast);
+      m_robotContainer.conveyor.frontMotor.setIdleMode(IdleMode.kCoast);
+      m_robotContainer.conveyor.backMotor.setIdleMode(IdleMode.kCoast);
+      m_robotContainer.shooter.motorA.setIdleMode(IdleMode.kCoast);
+      m_robotContainer.shooter.motorB.setIdleMode(IdleMode.kCoast);
```
Comment (balghane) [inline src/main/java/frc/robot/Robot.java]: Conveyor and shooter rollers we can skip. Just pivots, elevator, and steering is fine

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -49,11 +50,25 @@ public void robotPeriodic() {
     CommandScheduler.getInstance().run();
   }
 
-  /** This function is called once each time the robot enters Disabled mode. */
+  /** This function is called once each time the robot enters disabled mode. */
   @Override
   public void disabledInit() {
+    if (m_autonomousCommand != null) {
+      m_autonomousCommand.cancel();
+    }
```
Comment (balghane) [inline src/main/java/frc/robot/Robot.java]: Don't think we need this

File: src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -49,11 +50,25 @@ public void robotPeriodic() {
...
     LimelightHelpers.getLatestResults(
         IntakeLimelightConstants.INTAKE_LIMELIGHT_NAME); // It takes 2.5-3s on first run
+
+    if (m_robotContainer.matchState.realMatch) {
```
Comment (balghane) [inline src/main/java/frc/robot/Robot.java]: These are backwards, we _**don't**_ want to do this in a real match

**Review by tails618 — APPROVED** [review]
Remember to format

**Review by allison324 — APPROVED** [review]

**Pull request merged**